packages/db/src/translate/executions.ts

```diff
@@ -0,0 +1,88 @@
+import Knex from 'knex';
+
+import { RecordDoesNotExist } from '@cumulus/errors';
+import { ExecutionRecord } from '@cumulus/types/api/executions';
+import Logger from '@cumulus/logger';
+import { PostgresExecution } from '../types/execution';
+import { ExecutionPgModel } from '../models/execution';
+import { CollectionPgModel } from '../models/collection';
+import { AsyncOperationPgModel } from '../models/async_operation';
+
+/**
+ * Translate execution record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} knex
+ *   Knex client
+ * @returns {PostgresExecutionRecord} - converted Execution
+ */
+export const translateApiExecutionToPostgresExecution = async (
+  dynamoRecord: ExecutionRecord,
+  knex: Knex
+): Promise<PostgresExecution> => {
+  const execution = new ExecutionPgModel();
+  const collection = new CollectionPgModel();
+  const asyncOperation = new AsyncOperationPgModel();
+  const logger = new Logger({ sender: '@cumulus/db/translate/executions' });
+
+  // Map old record to new schema.
+  const translatedRecord: PostgresExecution = {
+    async_operation_cumulus_id: (
+      dynamoRecord.asyncOperationId ? await asyncOperation.getRecordCumulusId(
+        knex,
+        { id: dynamoRecord.asyncOperationId }
+      ) : undefined
+    ),
+    status: dynamoRecord.status,
+    tasks: JSON.stringify(dynamoRecord.tasks),
+    error: JSON.stringify(dynamoRecord.error),
+    arn: dynamoRecord.arn,
+    duration: dynamoRecord.duration,
+    original_payload: JSON.stringify(dynamoRecord.originalPayload),
+    final_payload: JSON.stringify(dynamoRecord.finalPayload),
+    workflow_name: dynamoRecord.type,
+    url: dynamoRecord.execution,
+    cumulus_version: dynamoRecord.cumulusVersion,
+  };
+
+  if (dynamoRecord.timestamp) {
```

Comment:
Nit: It occurs to me that you could handle all of these date fields with a ternary rather than `if` blocks

```
timestamp: dynamoRecord.timestamp ? new Date(Number(dynamoRecord.timestamp)) : undefined,
```

Also I'm curious about the use of the `Number` constructor here? Does this fail without it?

Reply:
Yup! Definitely. How about if I put all the date-related fields in the main `PostgresExecution` definition but leave the two foreign key fields in `if` blocks? Or I could move those to separate functions to be used in similar ternaries if ya like. My latest push shows it like I'm suggesting.

Reply:
Oh and you're totally right. the `Number` casting can be removed. I had it there because the test execution I was using before discovering the `fakeExecutionFactoryV2` had that field as a string.